sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/cryptography/BlobDecryptionPolicy.java

```diff
@@ -0,0 +1,489 @@
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.ShortBufferException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.security.GeneralSecurityException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.azure.storage.blob.cryptography.EncryptionConstants.ENCRYPTION_BLOCK_SIZE;
+
+public class BlobDecryptionPolicy implements HttpPipelinePolicy {
+
+    private final ClientLogger logger = new ClientLogger(BlobDecryptionPolicy.class);
+
+    /**
+     * The {@link IKeyResolver} used to select the correct key for decrypting existing blobs.
+     */
+    private final IKeyResolver keyResolver;
+
+    /**
+     * An object of type {@link IKey} that is used to wrap/unwrap the content key during encryption.
+     */
+    private final IKey keyWrapper;
+
+    /**
+     * Initializes a new instance of the {@link BlobDecryptionPolicy} class with the specified key and resolver.
+     * <p>
+     * If the generated policy is intended to be used for encryption, users are expected to provide a key at the
+     * minimum. The absence of key will cause an exception to be thrown during encryption. If the generated policy is
+     * intended to be used for decryption, users can provide a keyResolver. The client library will - 1. Invoke the key
+     * resolver if specified to get the key. 2. If resolver is not specified but a key is specified, match the key id on
+     * the key and use it.
+     *
+     * @param key An object of type {@link IKey} that is used to wrap/unwrap the content encryption key.
+     * @param keyResolver The key resolver used to select the correct key for decrypting existing blobs.
+     */
+    BlobDecryptionPolicy(IKey key, IKeyResolver keyResolver) {
+        if (key == null && keyResolver == null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException("Key and KeyResolver cannot both be null"));
+        }
+        this.keyWrapper = key;
+        this.keyResolver = keyResolver;
+    }
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        // 1. Expand the range of download for decryption
+        String rangeHeader = "x-ms-range";
+        HttpHeaders requestHeaders = context.getHttpRequest().getHeaders();
+        EncryptedBlobRange encryptedRange = EncryptedBlobRange.getEncryptedBlobRangeFromHeader(
+            requestHeaders.getValue(rangeHeader));
+
+        // Assumption: Download is the only API on an encrypted client that sets x-ms-range
+        // Only set the x-ms-range header if it already exists
```

Comment:
Another option here is checking the context. The client would have to insert a "decrypt: true" key/value into the context and then this policy could check for that.